src/formatDate.js

```diff
@@ -8,7 +8,33 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateParts = {};
+  const newDateParts = [];
+  const split = fromFormat.pop();
+  const newSplit = toFormat.pop();
```

Comment:
Using `pop()` on `toFormat` modifies the array destructively, which may lead to unexpected behavior if `toFormat` is reused elsewhere. Consider using `slice(-1)[0]` to get the last element without modifying the array.